strings/base_error.h

```diff
@@ -329,6 +336,13 @@ WINRT_EXPORT namespace winrt
         hresult_canceled(take_ownership_from_abi_t, winrt::impl::slim_source_location const& sourceInformation = winrt::impl::slim_source_location::current()) noexcept : hresult_error(impl::error_canceled, take_ownership_from_abi, sourceInformation) {}
     };
 
+    struct non_originating_hresult_canceled : hresult_error
+    {
+        non_originating_hresult_canceled(winrt::impl::slim_source_location const& sourceInformation = winrt::impl::slim_source_location::current()) noexcept : hresult_error(impl::error_canceled, hresult_error::avoid_originate, sourceInformation) {}
```

Comment:
Instead of a different struct I'm wondering if it would be cleaner to continue using `hresult_canceled` and fill in the `hresult_error::avoid_originate` parameter when it is appropriate.